pkg/controller/scyllacluster/resource_test.go

```diff
@@ -1876,6 +1876,380 @@ func TestMakeJobs(t *testing.T) {
+										},
+									},
+								},
+							},
+						},
+					},
+					PodAffinity: &corev1.PodAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+							{
+								LabelSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										"app.kubernetes.io/name": "scylla",
+										"scylla/rack":            "rack",
+									},
+								},
+								TopologyKey: "kubernetes.io/hostname",
+							},
+						},
+					},
+					PodAntiAffinity: &corev1.PodAntiAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+							{
+								LabelSelector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{
+										"scylla-operator.scylladb.com/node-job-type": "Cleanup",
+									},
+								},
+								TopologyKey: "kubernetes.io/hostname",
+							},
+						},
+					},
+					Tolerations: []corev1.Toleration{
+						{
+							Key:      "role",
+							Operator: corev1.TolerationOpEqual,
+							Value:    "scylla-clusters",
+							Effect:   corev1.TaintEffectNoSchedule,
+						},
+					},
+				}
+				cluster.Spec.Datacenter.Racks = append(cluster.Spec.Datacenter.Racks,
+					scyllav1.RackSpec{
+						Name: "rack-2",
+						Storage: scyllav1.StorageSpec{
+							Capacity: "1Gi",
+						},
+						Members: 1,
+						Placement: &scyllav1.PlacementSpec{
+							NodeAffinity: &corev1.NodeAffinity{
+								RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+									NodeSelectorTerms: []corev1.NodeSelectorTerm{
+										{
+											MatchExpressions: []corev1.NodeSelectorRequirement{
+												{
+													Key:      "topology.kubernetes.io/zone",
+													Operator: corev1.NodeSelectorOpIn,
+													Values:   []string{"zone-3", "zone-4"},
+												},
+											},
+										},
+									},
+								},
+							},
+							PodAffinity: &corev1.PodAffinity{
+								RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+									{
+										LabelSelector: &metav1.LabelSelector{
+											MatchLabels: map[string]string{
+												"app.kubernetes.io/name": "scylla",
+												"scylla/rack":            "rack-2",
+											},
+										},
+										TopologyKey: "kubernetes.io/hostname",
+									},
+								},
+							},
+							PodAntiAffinity: &corev1.PodAntiAffinity{
+								RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
+									{
+										LabelSelector: &metav1.LabelSelector{
+											MatchLabels: map[string]string{
+												"scylla-operator.scylladb.com/node-job-type": "Cleanup",
+											},
+										},
+										TopologyKey: "kubernetes.io/hostname",
+									},
+								},
+							},
+							Tolerations: []corev1.Toleration{
+								{
+									Key:      "role",
+									Operator: corev1.TolerationOpEqual,
+									Value:    "scylla-clusters",
+									Effect:   corev1.TaintEffectNoSchedule,
+								},
+							},
```

Comment:
please make the different from rack[0] as well

Reply:
done